design/fips-compliance.md

```diff
@@ -0,0 +1,213 @@
+ |----------------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+ | Release only `goboring/golang` version | - Customers use the client and don't require FIPS compliancy, will now miss libraries that are available in `golang` but are not available in `goboring/golang`                                                                                                                                                                                                       |
+ | Release both versions                  | <ul><li>One more Dockerfile to maintain. Any change made to the original Dockerfile should be done also to `Dockerfile.fips` which can be easily forgotten</li><li>Another version to test. Our UTs will stay the same but we will need to run integration tests on both images, in order to release them. This adds complexity to our tests and doubles the tests run-time</li></ul> |
+
+From the table above, I think that we should release only the FIPS compliant
+version. The advantages that customers will get by using the `golang` image
+as the base image is not worth the maintenance burden that releasing both
+versions will bring to the development team.
+
+### Red Hat image
+
+We will also release a FIPS compliant Red Hat image of the `conjur-authn-k8s
+-client` to our Red Hat registry. Before we do that, we should create a 
+`conjur-authn-k8s-client` FIPS project in the RH container registry.
+
+TBD - how do we create the FIPS project in the RH container registry?
+
+### Performance
+
+Same SLA of performance should be kept. As of today, we don't have any
+performance tests and an SLA (that I know of). We should decide on one and
+verify that the change to GoBoring doesn't impact the performance.
+
+We should write a performance test and run it twice - once on the `master
+` image and once on the new FIPS-compliance image - and verify that the
+ performance is not affected by the change. 
+
+## Test plan
+
+Before we dig into which tests should run, we should decide where, and how, they
+should run. Currently, the `conjur-authn-k8s-client` have UTs in its project
+and a vanilla test that runs in `kubernetes-conjur-demo`. The `kubernetes-conjur-demo`
+is triggered only when we push code to that repository. 
+
+Even while putting aside the fact that we have only a vanilla test, our current
+ test flow is still not optimal. 
+ Let's propose some options to improve our flow, and decide on the best one.
+
+### Trigger `kubernetes-conjur-demo` to run after every build of `conjur-authn-k8s-client`
```

Comment:
I would discourage use of the demo as an integration test of this authenticator. I think we can have all of it included in this repo if we tried (maybe using [KinD](https://github.com/kubernetes-sigs/kind) and Conjur OSS/Helm).

Reply:
+1 for using KinD for lightweight, realistic e2e tests.